packages/curve-ui-kit/.storybook/main.ts

```diff
@@ -0,0 +1,77 @@
+import { join, dirname } from 'path'
+import type { StorybookConfig } from '@storybook/nextjs'
+import path from 'path'
+
+/**
+ * This function is used to resolve the absolute path of a package.
+ * It is needed in projects that use Yarn PnP or are set up within a monorepo.
+ */
+function getAbsolutePath(packageName: string) {
+  return dirname(require.resolve(join(packageName, 'package.json')))
+}
+
+const config: StorybookConfig = {
+  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|mjs|mts|ts|tsx)'],
+  staticDirs: ['../public'],
+  addons: [
+    getAbsolutePath('@storybook/addon-essentials'),
+    getAbsolutePath('@chromatic-com/storybook'),
+    getAbsolutePath('@storybook/addon-interactions'),
+    getAbsolutePath('@storybook/addon-themes'),
+    getAbsolutePath('@storybook/addon-a11y'),
+  ],
+
+  framework: {
+    name: getAbsolutePath('@storybook/nextjs'),
+    options: {},
+  },
+
+  core: {
+    disableTelemetry: true,
+  },
+
+  docs: {},
+  typescript: {},
+
+  // TODO: wait for fix https://github.com/storybookjs/storybook/issues/12129
```

Comment:
this issue is from 2020 :dizzy_face:

Reply:
Unfortunately, so, that's why I'm using one of the proposed solutions 😭
https://github.com/storybookjs/storybook/issues/12129#issuecomment-1486722918